test/tools/schema_validator/test/schema_validator_tests.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -e
+
+# Router_check_tool binary path
+PATH_BIN="${TEST_SRCDIR}/envoy"/test/tools/schema_validator/schema_validator_tool
+
+# Config json path
+PATH_CONFIG="${TEST_SRCDIR}/envoy"/test/tools/schema_validator/test/config
+
+# No errors
+"${PATH_BIN}" "-c" "${PATH_CONFIG}/lds.yaml" "-t" "discovery_response"
+
+# No errors with deprecation and WiP checking
+"${PATH_BIN}" "-c" "${PATH_CONFIG}/lds.yaml" "-t" "discovery_response" "--fail-on-wip" \
+  "--fail-on-deprecated"
+
+# No errors without fail on deprecated
+"${PATH_BIN}" "-c" "${PATH_CONFIG}/lds_deprecated.yaml" "-t" "discovery_response"
+
+# Fail on deprecated
+BAD_CONFIG_OUTPUT=$("${PATH_BIN}" "-c" "${PATH_CONFIG}/lds_deprecated.yaml" "-t" \
+  "discovery_response" "--fail-on-deprecated" 2>&1) ||
+  echo "${BAD_CONFIG_OUTPUT:-no-output}"
+if [[ "${BAD_CONFIG_OUTPUT}" != *"Using deprecated option 'envoy.config.listener.v3.FilterChain.use_proxy_proto' from file listener_components.proto"* ]]; then
+  exit 1
+fi
+
+# Unknown field
+BAD_CONFIG_OUTPUT=$("${PATH_BIN}" "-c" "${PATH_CONFIG}/lds_unknown.yaml" "-t" \
+  "discovery_response" 2>&1) ||
+  echo "${BAD_CONFIG_OUTPUT:-no-output}"
+if [[ "${BAD_CONFIG_OUTPUT}" != *"reason INVALID_ARGUMENT:foo: Cannot find field."* ]]; then
+  exit 1
+fi
+
+# Invalid type struct URL cases
+BAD_CONFIG_OUTPUT=$("${PATH_BIN}" "-c" "${PATH_CONFIG}/lds_invalid_typed_struct.yaml" "-t" \
```

Comment:
Vague apprehension about introducing another shell script testing script. Is there a reason we couldn't have made this a gtest in C++? I kind of feel hotrestart is the way it is as we need to do a bunch of full binary and environmental things. Is that true here?

Reply:
I did this since I figured it would be better to test the real deal. I don't feel strongly about it, so if you prefer I can just make the test a thin wrapper around the main by passing in arguments manually. However, unless you want me to merge the other PR in, I would really prefer to merge this and iterate on the next PR? Or we can just do them both together and I will merge the next set of changes into this one and we can review it all together. What do you think?